### ISSUE TITLE
Add `consistent` option to `comma-dangle`

### ISSUE THREAD

**Issue by BridgeAR** [eslint/eslint#10350]
**What rule do you want to change?**

comma-dangle

**Does this change cause the rule to produce more or fewer warnings?**

That depends.

**How will the change be implemented? (New option, new default behavior, etc.)?**

This is a feature request from Node.js: it would be nice to add a `consistent` option to `comma-dangle` that checks if the whole file has a consistent style of either always having trailing commas or having no trailing commas at all. With the current options something like this is not possible.

Refs: https://github.com/nodejs/node/pull/20703#issuecomment-388644134

**Please provide some example code that this change will affect:**

```js
const a = [
  1,
  2,
]

const b = [
  1,
  2
]
```

```js
const a = {
  a: 1,
  b: 2,
}

const b = {
  a: 1,
  b: 2
}
```

**What does the rule currently do for this code?**

Using the `only-multiline` option results in no error. Using the `always-multiline` always reports errors.

**What will the rule do after it's changed?**

It should report an issue as long as both styles are used in a single file. If the file has either a consistent trailing comma style or no trailing commas, it should not report an issue.

**Comment by ilyavolodin** [comment]
Sounds like a reasonable option to add.
